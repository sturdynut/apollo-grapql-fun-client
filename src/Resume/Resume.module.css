.container {
  text-align: center;
}

.listItem {
  padding: 1rem;
  border-bottom: 1px solid #000;
}

.listItemHighlighted {
  background-color: red;
}
