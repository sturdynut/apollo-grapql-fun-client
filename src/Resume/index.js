import { gql } from "apollo-boost";
import { useQuery } from "@apollo/react-hooks";
import React, { useState } from "react";
import cx from "classnames";

import styles from "./Resume.module.css";

const query = gql`
  # Write your query or mutation here
  {
    roles {
      title
      employers {
        name
        startDate
        endDate
        accomplishments
      }
    }
  }
`;

function Resume() {
  // const [isActive, setIsActive] = useState(false);
  const { data, loading, error } = useQuery(query);

  console.log({
    data,
    loading,
    error
  });

  if (loading) return "Loading...";

  const { roles } = data;

  return (
    <div className={styles.container}>
      {roles.map(role => (
        <div>
          <h3>{role.title}</h3>
        </div>
      ))}
    </div>
  );
}

export default Resume;

// <div className={styles.container}>
//       <ul>
//         <li
//           className={cx([
//             styles.listItem,
//             {
//               [styles.listItemHighlighted]: isActive
//             }
//           ])}
//         >
//           Active
//         </li>
//         <li className={styles.listItem}>Inactive</li>
//       </ul>
//       <button
//         onClick={() => {
//           setIsActive(!isActive);
//         }}
//       >
//         Oh haaay
//       </button>
//     </div>
