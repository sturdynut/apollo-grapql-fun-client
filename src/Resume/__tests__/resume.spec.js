import { ApolloProvider } from "@apollo/react-hooks";
import ApolloClient from "apollo-boost";
import React from "react";
import { mount } from "enzyme";
import Resume from "../";

const client = new ApolloClient({
  uri: "http://localhost:4000/graphql"
});

describe("<Resume />", () => {
  test("renders as expected", () => {
    const wrapper = mount(
      <ApolloProvider client={client}>
        <Resume />
      </ApolloProvider>
    )
      .children()
      .find("Resume");
    expect(wrapper).toMatchSnapshot();
  });
});
