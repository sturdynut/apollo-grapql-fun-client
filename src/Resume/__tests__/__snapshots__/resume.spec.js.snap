// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Resume /> renders as expected 1`] = `
<Resume>
  Loading...
</Resume>
`;
